### PR TITLE
Virtualization docs fixes during ROSA review

### PR DESCRIPTION
**Pull request by mburke5678**
Fixing various errors in the Applications docs as I find them during the ROSA content port. Mostly formatting, wording, and such.

Previews:
[Installing VirtIO drivers from a container disk added as a SATA CD drive](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/creating_vms_custom/virt-installing-qemu-guest-agent#virt-adding-container-disk-as-cd_virt-installing-qemu-guest-agent) -- Added `-n namespace` to command.
[Adding a key when creating a VM by using the command line](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-accessing-vm-ssh#virt-adding-public-key-cli_static-key) and [Enabling dynamic key injection by using the command line](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-accessing-vm-ssh#virt-adding-public-key-cli_static-key) -- Added namespace to code example in step 3 to match namespace in step 1 code block.
[Adding a virtual machine to a service mesh](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/vm_networking/virt-connecting-vm-to-service-mesh#virt-adding-vm-to-service-mesh_virt-connecting-vm-to-service-mesh) -- Fixed YAML spacing in the code example in step 3.
[Adding a vTPM device to a virtual machine](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-using-vtpm-devices#virt-adding-vtpm-to-vm_virt-using-vtpm-devices) -- Added `-n namespace` to command in step 1.
[Configuring a storage class for custom boot source updates](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/storage/virt-automatic-bootsource-updates#virt-configuring-storage-class-bootsource-update_virt-automatic-bootsource-updates) -- Added required parameters to code example. [ ] QE has approved this change.  Fixed spacing in code block in step 3a.
[Connecting to a VM attached to a secondary network by using SSH](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-accessing-vm-ssh#virt-connecting-secondary-network-ssh_virt-accessing-vm-ssh) -- Added `-n namespace` to command.
[Creating a Linux bridge NAD by using the command line](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/vm_networking/virt-connecting-vm-to-linux-bridge#virt-creating-linux-bridge-nad-cli_virt-connecting-vm-to-linux-bridge) -- Added required quotes to values. [ ] QE has approved this change.
[Creating a service by using the command line](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-accessing-vm-ssh#virt-creating-service-cli_virt-accessing-vm-ssh) -- Added required parameters to step 3. [ ] QE has approved this change.
[Creating a VM from a VirtualMachine manifest](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/creating_vms_rh/virt-creating-vms-from-cli#virt-creating-vm-cli_virt-creating-vms-cli) -- Added `-n <namespace>` to code example in step 3
[Creating a VM from a container disk by using the web console](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/creating_vms_custom/virt-creating-vms-from-container-disks#virt-creating-vm-custom-image-web_virt-creating-vms-from-container-disks) -- Corrected the web console steps.
[Customizing the storage profile](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/storage/virt-configuring-storage-profile#virt-customizing-storage-profile_virt-configuring-storage-profile) -- Fixed punctuation in step 1.
[Disabling TLS for a container registry](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/creating_vms_custom/virt-creating-vms-from-container-disks#virt-disabling-tls-for-registry_virt-creating-vms-from-container-disks) -- Added example of how to edit the file in Step 1.
[Editing a boot order list in the YAML configuration file](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-edit-boot-order#virt-edit-boot-order-yaml-web_virt-edit-boot-order) -- Added `-n namespace` to command.
[Editing a virtual machine by using the command line](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-edit-vms#virt-editing-vm-cli_virt-edit-vms) -- Added `-n namespace` to command.
[Enabling dynamic SSH key injection by using the web console](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-accessing-vm-ssh#virt-editing-vm-dynamic-key-injection_dynamic-key) -- Updated procedure step 3 to match what is in console.
[Enabling preallocation for a data volume](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/storage/virt-using-preallocation-for-datavolumes#virt-enabling-preallocation-for-dv_virt-using-preallocation-for-datavolumes) -- Added required parameters to code block. [ ] QE has approved this change.
[Hot plugging a bridge network interface using the CLI](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/vm_networking/virt-hot-plugging-network-interfaces#virt-hot-plugging-bridge-network-interface_virt-hot-plugging-network-interfaces) -- Added `-n namespace` to command in step 1.
[Initiating live migration by using the command line](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/live_migration/virt-initiating-live-migration#virt-initiating-vm-migration-cli_virt-initiating-live-migration) -- Added `-n namespace` to command.
[Pause a virtual machine](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-controlling-vm-states#virt-pausing-vm-web_virt-controlling-vm-states) -- New file copied from [virt-starting-vm-web.adoc](https://github.com/openshift/openshift-docs/pull/69051/files#diff-8cb9d109052efd28a6ee8492466998313d6ed6fd2138f268473f8d79c924d581) and changed start to pause.
Initiating live migration by using the command line -- Added `-n namespace` to command in verification.
[PXE booting with a specified MAC address](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/advanced_vm_management/virt-configuring-pxe-booting#virt-pxe-booting-with-mac-address_pxe-booting) -- Added verification heading.
[Restarting a virtual machine](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-controlling-vm-states#virt-restarting-vm-web_virt-controlling-vm-states) --  Updated substeps under step 3 to match web console.
[Starting a virtual machine](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-controlling-vm-states#virt-starting-vm-web_virt-controlling-vm-states) -- Updated substeps under step 3 to match web console.
[Stopping a virtual machine](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-controlling-vm-states#virt-stopping-vm-web_virt-controlling-vm-states) --  Updated substeps under step 3 to match web console.
[Unpausing a virtual machine](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-controlling-vm-states#virt-unpausing-vm-web_virt-controlling-vm-states) -- Updated substeps under step 3 to match web console.
Post-install ->Network configuration  -> Demoted assembly-level next step to module level in [Creating a Linux bridge NAD by using the web console](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/post_installation_configuration/virt-post-install-network-config#virt-creating-linux-bridge-nad-web_virt-post-install-network-config) and [Configuring SR-IOV network devices](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/post_installation_configuration/virt-post-install-network-config#nw-sriov-configuring-device_virt-post-install-network-config).
[Controlling virtual machine states](https://69051--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-controlling-vm-states) -- Reordered modules and added new module for pausing. [Current docs](https://docs.openshift.com/container-platform/4.14/virt/virtual_machines/virt-controlling-vm-states.html).

### CODE REVIEW

File: modules/virt-configuring-storage-class-bootsource-update.adoc [hunk 1]
```diff
@@ -39,9 +39,13 @@ spec:
       template:
         spec:
           storageClassName: <new_storage_class> <1>
+      schedule: "0 */12 * * *" <2>
+      managedDataSource: <data_source> <3>
 #...
 ----
 <1> Define the storage class.
+<2> Required: Schedule for the job specified in cron format.
+<3> Required: For the custom image to be detected as an available boot source, the name of the image's `managedDataSource` must match the name of the template's `DataSource`, which is found under `spec.dataVolumeTemplates.spec.sourceRef.name` in the VM template YAML file.
```
Comment (mburke5678) [inline modules/virt-configuring-storage-class-bootsource-update.adoc:48]: **QE REVIEW NEEDED** Please verify that these parameters are required and the definitions are correct.

File: modules/virt-creating-linux-bridge-nad-cli.adoc [hunk 2]
```diff
@@ -34,9 +34,9 @@ metadata:
 spec:
   config: '{
     "cniVersion": "0.3.1",
-    "name": bridge-network, <3>
-    "type": cnv-bridge, <4>
-    "bridge": bridge-interface, <5>
+    "name": "bridge-network", <3>
+    "type": "cnv-bridge", <4>
+    "bridge": "bridge-interface", <5>
```
Comment (mburke5678) [inline modules/virt-creating-linux-bridge-nad-cli.adoc:39]: QE REVIEW NEEDED Please verify that these parameter values require quotes.

File: modules/virt-creating-service-cli.adoc [hunk 3]
```diff
@@ -55,9 +55,14 @@ spec:
...
     special: key <1>
   type: NodePort <2>
+  ports: <3>
+    protocol: TCP
+    port: 80
+    targetPort: 9376
 ----
 <1> Specify the label that you added to the `spec.template.metadata.labels` stanza of the `VirtualMachine` manifest.
 <2> Specify `ClusterIP`, `NodePort`, or `LoadBalancer`.
+<3> Specifies a collection of network ports and protocols that you want to expose from the virtual machine.
```
Comment (mburke5678) [inline modules/virt-creating-service-cli.adoc:65]: **QE REVIEW NEEDED** Please verify that these parameters are required and the definitions are correct.

File: modules/virt-disabling-tls-for-registry.adoc [hunk 4]
```diff
@@ -14,8 +14,14 @@ You can disable TLS (transport layer security) for one or more container registr
...
 
-* Edit the `HyperConverged` custom resource and add a list of insecure registries to the `spec.storageImport.insecureRegistries` field.
+* Use the following command to edit the `HyperConverged` custom resource and add a list of insecure registries to the `spec.storageImport.insecureRegistries` field.
 +
+[source,terminal]
+----
+$ oc edit Hyperconverged kubevirt-hyperconverged -n {CNVNamespace}
+----
++
+.Example `HyperConverged` custom resource
```
Comment (mburke5678) [inline modules/virt-disabling-tls-for-registry.adoc:24]: **NOTE TO REVIEWER** The `-n {CNVNamespace}` is used throughout the Virtualization book. I didn't update here; would likely be best as a separate PR, if required.

File: modules/virt-pausing-vm-web.adoc [hunk 5]
```diff
@@ -0,0 +1,32 @@
+// Module included in the following assemblies:
```
Comment (mburke5678) [inline modules/virt-pausing-vm-web.adoc:1]: New file copied from [virt-starting-vm-web.adoc](https://github.com/openshift/openshift-docs/pull/69051/files#diff-8cb9d109052efd28a6ee8492466998313d6ed6fd2138f268473f8d79c924d581) and changed _start_ to _pause_.

**Comment by kalexand-rh** [comment]
The `branch/enterprise-4.16` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.15. And any PR going into `main` must also target the latest version branch (enterprise-4.16).

If the update in your PR does NOT apply to version 4.16 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

File: modules/virt-configuring-storage-class-bootsource-update.adoc [hunk 6]
```diff
@@ -39,9 +39,13 @@ spec:
...
         spec:
           storageClassName: <new_storage_class> <1>
-#...
+      schedule: "0 */12 * * *" <2>
+      managedDataSource: <data_source> <3>
+# ...
 ----
 <1> Define the storage class.
+<2> Required: Schedule for the job specified in cron format.
+<3> Required: For the custom image to be detected as an available boot source, the name of the image's `managedDataSource` must match the name of the template's `DataSource`, which is found under `spec.dataVolumeTemplates.spec.sourceRef.name` in the VM template YAML file.
```
Comment (ousleyp) [inline modules/virt-configuring-storage-class-bootsource-update.adoc:48]: Suggestion to potentially improve clarity:
```suggestion
<3> Required: For the custom image to be detected as an available boot source, the value of the `managedDataSource` field must match the value of the `spec.dataVolumeTemplates.spec.sourceRef.name` field in the VM template YAML file.
```

File: modules/virt-disabling-tls-for-registry.adoc (hunk 4 above)
Comment (ousleyp) [inline modules/virt-disabling-tls-for-registry.adoc:24]:
```suggestion
. Open the `HyperConverged` CR in your default editor by running the following command:
+
[source,terminal,subs="attributes+"]
----
$ oc edit hyperconverged kubevirt-hyperconverged -n {CNVNamespace}
----

. Add a list of insecure registries to the `spec.storageImport.insecureRegistries` field.
+
.Example `HyperConverged` custom resource
```

and then after the example, I'd throw this in to match our other modules:
```
. Save the file.
```

Comment (ousleyp) [reply modules/virt-disabling-tls-for-registry.adoc:24]: This should work with `subs="attributes+"`; I'll make a note to look for instances that are missing this

File: modules/virt-edit-boot-order-yaml-web.adoc [hunk 7]
```diff
@@ -43,7 +43,4 @@ interfaces:
 <1> The boot order value specified for the disk.
 <2> The boot order value specified for the network interface controller.
 
-. Save the YAML file.
-
-
-. Click *reload the content* to apply the updated boot order values from the YAML file to the boot order list in the web console.
```
Comment (ousleyp) [inline modules/virt-edit-boot-order-yaml-web.adoc:49]: Does this removal need QE?

File: modules/virt-enabling-preallocation-for-dv.adoc [hunk 8]
```diff
@@ -24,7 +24,11 @@ spec:
   source: <1>
     pvc:
       preallocation: true <2>
+      name: <pvc_name> <3>
+      namespace: <pvc_namespace> <4>
 # ...
 ----
 <1> All CDI source types support preallocation. However, preallocation is ignored for cloning operations.
 <2> The default value is `false`.
+<3> Specify the names of the source PVC.
```
Comment (ousleyp) [inline modules/virt-enabling-preallocation-for-dv.adoc:33]:
```suggestion
<3> Specify the name of the source PVC.
```
?

File: modules/virt-pausing-vm-web.adoc [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+//
+// * virt/virtual_machines/virt-controlling-vm-states.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="virt-pausing-vm-web_{context}"]
+= Pause a virtual machine
```
Comment (ousleyp) [inline modules/virt-pausing-vm-web.adoc:7]:
```suggestion
= Pausing a virtual machine
```

File: modules/virt-pxe-booting-with-mac-address.adoc [hunk 10]
```diff
@@ -145,6 +145,7 @@ $ virtctl vnc vmi-pxe-boot
 ----
 $ virtctl console vmi-pxe-boot
 ----
+.Verification
```
Comment (ousleyp) [inline modules/virt-pxe-booting-with-mac-address.adoc:148]: The `Verification` heading isn't rendering for some reason. Maybe it just needs a line break before it?

File: modules/virt-restarting-vm-web.adoc [hunk 11]
```diff
@@ -23,14 +23,12 @@ To avoid errors, do not restart a virtual machine while it has a status of *Impo
...
 * To view comprehensive information about the selected virtual machine before
 you restart it:
 
 .. Access the *VirtualMachine details* page by clicking the name of the virtual
 machine.
 
-.. Click *Actions* -> *Restart*.
-
-. In the confirmation window, click *Restart* to restart the virtual machine.
+.. Click *Actions* -> *Start*.
```
Comment (ousleyp) [inline modules/virt-restarting-vm-web.adoc:34]: Are these changes correct/intentional?

**Review by ousleyp — COMMENTED** [review]
Thanks so much for making these fixes, @mburke5678 ! Aside from a few comments, these are looking good.

File: modules/virt-edit-boot-order-yaml-web.adoc (hunk 7 above)
Comment (mburke5678) [reply modules/virt-edit-boot-order-yaml-web.adoc:49]: @ousleyp Yeah. Probably not a bad idea.

File: modules/virt-restarting-vm-web.adoc (hunk 11 above)
Comment (mburke5678) [reply modules/virt-restarting-vm-web.adoc:34]: I'm going to say that it is. Perhaps you could have QE verify?

File: modules/virt-restarting-vm-web.adoc [hunk 12]
```diff
@@ -23,14 +23,12 @@ To avoid errors, do not restart a virtual machine while it has a status of *Impo
 
 * To stay on this page, where you can perform actions on multiple virtual machines:
 
-.. Click the Options menu {kebab} located at the far right end of the row.
+.. Click the Options menu {kebab} located at the far right end of the row and click *Resume VirtualMachine*.
```
Comment (ousleyp) [inline modules/virt-restarting-vm-web.adoc:26]:
```suggestion
.. Click the Options menu {kebab} located at the far right end of the row and click *Restart*.
```

File: modules/virt-restarting-vm-web.adoc (hunk 11 above)
Comment (ousleyp) [inline modules/virt-restarting-vm-web.adoc:34]:
```suggestion
.. Click *Actions* -> *Restart*.
```

**Review by ousleyp — COMMENTED** [review]
Submitting two comments regarding modules/virt-restarting-vm-web.adoc after a quick look at the UI

File: modules/virt-restarting-vm-web.adoc (hunk 11 above)
Comment (ousleyp) [reply modules/virt-restarting-vm-web.adoc:34]: Some of it looks right (removing the confirmation window step) but I think there are some necessary changes. Please see https://github.com/openshift/openshift-docs/pull/69051#pullrequestreview-1940555314 

and yes, QE is needed here too

Comment (mburke5678) [reply modules/virt-restarting-vm-web.adoc:34]: @ousleyp For this one, I am seeing the **Restart** option grated-out. The user can only click **Start**. Is this a bug?? 

image

File: modules/virt-restarting-vm-web.adoc (hunk 12 above)
Comment (mburke5678) [reply modules/virt-restarting-vm-web.adoc:26]: @ousleyp For this one, I no longer see **Resume VirtualMachine**. However, I see that **Restart** is grayed out. The user needs to click **Start**?? (This is a 4.14.16 cluster.)

image

File: modules/virt-restarting-vm-web.adoc (hunk 11 above)
Comment (ousleyp) [reply modules/virt-restarting-vm-web.adoc]: my other comment applies here too

Comment (mburke5678) [reply modules/virt-restarting-vm-web.adoc]: @ousleyp Thanks for verifying!

**Comment by mburke5678** [comment]
@kbidarkar As part of the project to port docs content from OCP to ROSA/OSD, I performed each of the procedures in the Virtualization book. I found various places where there were changes, mistakes, etc. There are a few changes that I would feel more comfortable if QE reviewed the changes, all are indicated with _**QE REVIEW NEEDED**_. Pan suggested this is something you could help with.  Please let me know if you can help. There is no hard deadline; but, I would like to merge these PRs, as they have been lingering for awhile.

File: modules/virt-creating-service-cli.adoc (hunk 3 above)
Comment (dshchedr) [reply modules/virt-creating-service-cli.adoc:65]: looks good, also there is an optional parameter
```
  ports:
    nodePort: 30000
```
if it is not set - its value will be allocated dynamically 
however not sure how important to say about it..

File: modules/virt-configuring-storage-class-bootsource-update.adoc [hunk 13]
```diff
@@ -76,4 +79,4 @@ $ oc patch storageclass <current_default_storage_class> -p '{"metadata": {"annot
 ----
 $ oc patch storageclass <new_storage_class> -p '{"metadata":{"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}' <1>
 ----
-<1> Replace `<new_storage_class>` with the `storageClassName` value that you added to the `HyperConverged` CR.
\ No newline at end of file
+<1> Replace `<new_storage_class>` with the `storageClassName` value that you added to the `HyperConverged` CR.
```
Comment (jpeimer) [inline modules/virt-configuring-storage-class-bootsource-update.adoc:82]: I think we need to edit some things at the beginning of the doc as well.

`Specify a new default storage class in the `HyperConverged` custom resource (CR).` - does not sound right. We need either a default storage class, or a specified in the HyperConverged CR, which should override the default.

In addition to that, since 4.15 we have a default-virt annotation
`storageclass.kubevirt.io/is-default-virt-class: "true"`
that will be preferred over the kubernetes default annotation
`storageclass.kubernetes.io/is-default-class":"true"`

This is a bit information about this annotation: https://github.com/kubevirt/community/blob/main/design-proposals/default-virt-storage-class.md
(please pay attention, in the design proposal there's an outdated annotation (different wording), but here I pasted the correct one)

File: modules/virt-customizing-storage-profile.adoc [hunk 14]
```diff
@@ -22,7 +22,7 @@ If you create a data volume and omit YAML attributes and these attributes are no
 
 .Procedure
 
-. Edit the storage profile. In this example, the provisioner is not recognized by CDI:
+. Edit the storage profile. In this example, the provisioner is not recognized by CDI.
 +
 [source,terminal,subs="attributes+"]
 ----
```
Comment (jpeimer) [inline modules/virt-customizing-storage-profile.adoc:28]: `$ oc edit storageprofile <storage_class>`

Because storageprofile is not a namespaced resource

File: modules/virt-enabling-preallocation-for-dv.adoc [hunk 15]
```diff
@@ -24,7 +24,11 @@ spec:
   source: <1>
     pvc:
       preallocation: true <2>
+      name: <pvc_name> <3>
+      namespace: <pvc_namespace> <4>
 # ...
 ----
 <1> All CDI source types support preallocation. However, preallocation is ignored for cloning operations.
```
Comment (jpeimer) [inline modules/virt-enabling-preallocation-for-dv.adoc:31]: `preallocation is ignored for cloning operations`, but the example yaml shows us the cloning from a PVC. 
Maybe this would be a better example
```
spec:
  source:
      http: 
            ....
```

@awels, you know this topic better than anyone else, could you please review this doc as well?

File: modules/virt-enabling-preallocation-for-dv.adoc [hunk 16]
```diff
@@ -24,7 +24,11 @@ spec:
...
 <2> The default value is `false`.
+<3> Specify the name of the source PVC.
+<4> Specify the namespace of the source PVC.
```
Comment (jpeimer) [inline modules/virt-enabling-preallocation-for-dv.adoc:34]: If we'll use  `source: http: ...`, this can be removed

**Review by jpeimer — CHANGES_REQUESTED** [review]

File: modules/virt-configuring-storage-class-bootsource-update.adoc (hunk 13 above)
Comment (mburke5678) [reply modules/virt-configuring-storage-class-bootsource-update.adoc:82]: @ousleyp Are you awar of this? It seem like a matter that should be addressed on its own. 

> In addition to that, since 4.15 we have a default-virt annotation
> storageclass.kubevirt.io/is-default-virt-class: "true"
> that will be preferred over the kubernetes default annotation
> storageclass.kubernetes.io/is-default-class":"true"

File: modules/virt-configuring-storage-class-bootsource-update.adoc [hunk 17]
```diff
@@ -39,9 +39,13 @@ spec:
       template:
         spec:
           storageClassName: <new_storage_class> <1>
-#...
+      schedule: "0 */12 * * *" <2>
```
Comment (akalenyu) [inline modules/virt-configuring-storage-class-bootsource-update.adoc:42]: this section is about changing the storage class for boot images. how come we suddenly want to include the schedule/managedDataSource field?

File: modules/virt-configuring-storage-class-bootsource-update.adoc (hunk 13 above)
Comment (akalenyu) [reply modules/virt-configuring-storage-class-bootsource-update.adoc:82]: virt sc was handled in https://github.com/openshift/openshift-docs/pull/70639

**Review by akalenyu — COMMENTED** [review]
specifically reviewing the virt sc / custom storage class changes
as you can see I am a bit confused, as this has more than wording changes

File: modules/virt-configuring-storage-class-bootsource-update.adoc (hunk 13 above)
Comment (jpeimer) [reply modules/virt-configuring-storage-class-bootsource-update.adoc:82]: Right, it was addressed separately, but if we have a section about the default storage class for the boot source images - we should mention here both virt default and kubernetes default.

virt default will be preferred, but it will not show the **(default)** part, when we do `oc get storageclass`

File: modules/virt-configuring-storage-class-bootsource-update.adoc (hunk 17 above)
Comment (mburke5678) [reply modules/virt-configuring-storage-class-bootsource-update.adoc:42]: @akalenyu  If I am reading this right, in order to add `spec.dataImportCronTemplates.spec.template.spec.storageClassName` parameter, we need to add the `schedule` and `managedDataSource` parameters. Am I doing something wrong? 

```
# Please edit the object below. Lines beginning with a '#' will be ignored,
# and an empty file will abort the edit. If an error occurs while saving this file will be
# reopened with the relevant failures.
#
# hyperconvergeds.hco.kubevirt.io "kubevirt-hyperconverged" was not valid:
# * spec.dataImportCronTemplates[0].spec.managedDataSource: Required value
# * spec.dataImportCronTemplates[0].spec.schedule: Required value
```

**Comment by mburke5678** [comment]
@awels Are you able to look at @jpeimer [comment above](https://github.com/openshift/openshift-docs/pull/69051#discussion_r1541978936)?

File: modules/virt-configuring-storage-class-bootsource-update.adoc (hunk 17 above)
Comment (akalenyu) [reply modules/virt-configuring-storage-class-bootsource-update.adoc:42]: ah you're right, this was an actual doc bug then

File: modules/virt-configuring-storage-class-bootsource-update.adoc (hunk 13 above)
Comment (jpeimer) [reply modules/virt-configuring-storage-class-bootsource-update.adoc:82]: @mburke5678 so will we address this as a separate issue?

File: modules/virt-enabling-preallocation-for-dv.adoc (hunk 15 above)
Comment (awels) [reply modules/virt-enabling-preallocation-for-dv.adoc:31]: I would probably use a registry source, since most of the other examples in the document use it as well. And yes pre-allocation with cloning is a bit muddied, so something else is definitely better.

File: modules/virt-configuring-storage-class-bootsource-update.adoc [hunk 18]
```diff
@@ -39,9 +39,13 @@ spec:
...
+      managedDataSource: <data_source> <3>
+# ...
 ----
 <1> Define the storage class.
+<2> Required: Schedule for the job specified in cron format.
+<3>  Required: For the custom image to be detected as an available boot source, the value of the `managedDataSource` field must match the value of the `spec.dataVolumeTemplates.spec.sourceRef.name` field in the VM template YAML file.
```
[the remark this answers is not in the document]
Comment (mburke5678) [reply modules/virt-configuring-storage-class-bootsource-update.adoc:48]: @awels Thanks for your feedback. I might have gotten the wording from elsewhere in the docs, [Enabling automatic updates for custom boot sources, callout 4](https://docs.openshift.com/container-platform/4.15/virt/storage/virt-automatic-bootsource-updates.html#virt-autoupdate-custom-bootsource_virt-automatic-bootsource-updates). 

> For the custom image to be detected as an available boot source, the name of the image's `managedDataSource` must match the name of the template's `DataSource`, which is found under `spec.dataVolumeTemplates.spec.sourceRef.name` in the VM template YAML file.

[the remark this answers is not in the document]
Comment (awels) [reply modules/virt-configuring-storage-class-bootsource-update.adoc:48]: Like I said it is technically correct, as the two need to match, but it becomes incorrect when you have multiple VM templates referencing the same dataSourceTemplate. The VM template must match the DS template, not the other way around. And the way it is worded right now it seems like the DS template must match the VM template, which is correct most of the time but in the one case I mention it might not be correct.

**Comment by kbidarkar** [comment]
@jpeimer and @dshchedr from QE have reviewed it. Apart from the comments left by them, it LGTM.  @mburke5678 FYI.

File: modules/virt-configuring-storage-class-bootsource-update.adoc (hunk 18 above)
[the remark this answers is not in the document]
Comment (mburke5678) [reply modules/virt-configuring-storage-class-bootsource-update.adoc]: Ugh. Sorry this is all fairly new to me. [One more attempt](https://github.com/openshift/openshift-docs/pull/69051/commits/e136dc4b7ef059e8f196027edfadf7a1ab3005a4)! 

> For the custom image to be detected as an available boot source, the value of the `spec.dataVolumeTemplates.spec.sourceRef.name` parameter in the VM template must this value.

File: modules/virt-enabling-preallocation-for-dv.adoc (hunk 15 above)
Comment (mburke5678) [reply modules/virt-enabling-preallocation-for-dv.adoc:31]: I am not sure what is the suggested change regarding the source.  

Also, should we strike this line?
```suggestion
<1> All CDI source types support preallocation. 
```

Comment (mburke5678) [reply modules/virt-enabling-preallocation-for-dv.adoc:31]: @awels Any thoughts here? ^^

Comment (mburke5678) [reply modules/virt-enabling-preallocation-for-dv.adoc:31]: @awels @jpeimer Do you have an example DataVolume CR that uses a registry source that we could use in the docs?

**Review by jpeimer — APPROVED** [review]
